### PR TITLE
Rebase Dawn

### PR DESCRIPTION
**Pull request by JiangYizhou**
1.Add Vulkan validation layers on Windows
 https://dawn-review.googlesource.com/c/dawn/+/11120
 Vulkan validation layer is defaultly enabled on Windows, to disable Vulkan
 validation layer, pass args "dawn_enable_vulkan_validation_layers=false"
2.Update naming for vertex state
 https://github.com/gpuweb/gpuweb/pull/469
 https://dawn-review.googlesource.com/c/dawn/+/13100

### CODE REVIEW

**Comment by JiangYizhou** [comment]
@haoxli This patch enables Vulkan validation layer for Aquarium. PTAL. Thx.
We have to pass 'dawn_enable_vulkan_validation_layers=false' when gn gen out\Release.

**Comment by Jiawei-Shao** [comment]
I still think integrating the vulkan validation layer into Aquarium's GN building system is a better solution.

File: DEPS [hunk 1]
```diff
@@ -22,6 +22,8 @@ vars = {
   'shaderc_revision': '65adcb504d024bb29f0ad8279b6a215d5f679832',
   'jsoncpp_revision': '571788934b5ee8643d53e5d054534abbe6006168',
   'jsoncpp_source_revision': '645250b6690785be60ab6780ce4b58698d884d11',
+  'vulkan-headers_revision':'5b44df19e040fca0048ab30c553a8c2d2cb9623e',
+  'vulkan-validation-layers_revision':'9fba37afae13a11bd49ae942bf82e5bf1098e381',
```
Comment (haoxli) [inline DEPS:26]: If we don't need Vulkan validation layers in Aquarium, just need override `dawn_enable_vulkan_validation_layers` with false in .gn. Not need add these dependencies.

File: build_overrides/vulkan_validation_layers.gni [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+vulkan_headers_dir = "//third_party/vulkan-headers"
+vvl_spirv_tools_dir = "//third_party/SPIRV-Tools"
+vvl_glslang_dir = "//third_party/glslang"
+
+# Subdirectories for generated files
+vulkan_data_subdir = "vulkandata"
+vulkan_gen_subdir = ""
+
+dawn_vulkan_validation_layers_dir = "//third_party/vulkan-validation-layers"
+dawn_enable_vulkan_validation_layers = is_win&&is_debug
```
Comment (haoxli) [inline build_overrides/vulkan_validation_layers.gni]: Override `dawn_vulkan_validation_layers_dir` and `dawn_enable_vulkan_validation_layers` in build_overrides/dawn.gni, not build_overrides/vulkan_validation_layers.gni.

File: README.md [hunk 3]
```diff
@@ -215,8 +215,8 @@ gclient sync
 # On windows, opengl, d3d12 and dawn backends are enabled by default.
 # On linux and macOS, opengl and dawn are enabled by default.
 # Enable or disable a specific platform, you can add 'enable_opengl', 'enable_d3d12', and 'enable_dawn' to gn args.
-# To build a release version, specify 'is_debug=false'.
-gn gen out/Release --args="is_debug=false"
+# To build a release version, specify 'is_debug=false dawn_enable_vulkan_validation_layers=false'.
+gn gen out/Release --args="is_debug=false dawn_enable_vulkan_validation_layers=false"
```
Comment (haoxli) [inline README.md]: We can remove `dawn_enable_vulkan_validation_layers=false` if we override it in .gn

File: DEPS [hunk 4]
```diff
@@ -87,6 +89,14 @@ deps = {
...
     'condition': 'dawn_standalone',
   },
+  'third_party/vulkan-headers': {
+    'url': '{chromium_git}/external/github.com/KhronosGroup/Vulkan-Headers@{vulkan-headers_revision}',
+    'condition': 'dawn_standalone',
+  },
+  'third_party/vulkan-validation-layers': {
+    'url': '{chromium_git}/external/github.com/KhronosGroup/Vulkan-ValidationLayers@{vulkan-validation-layers_revision}',
+    'condition': 'dawn_standalone',
+  },
```
Comment (haoxli) [inline DEPS:99]: ditto

File: DEPS (hunk 1 above)
Comment (JiangYizhou) [reply DEPS:26]: Vulkan validation layer is quite useful for Aquarium to catch errors in dawn_vulkan backend.

File: build_overrides/vulkan_validation_layers.gni (hunk 2 above)
Comment (JiangYizhou) [reply build_overrides/vulkan_validation_layers.gni]: Done

File: README.md (hunk 3 above)
Comment (JiangYizhou) [reply README.md]: Done

File: DEPS (hunk 4 above)
Comment (JiangYizhou) [reply DEPS:99]: Done

File: src/aquarium-optimized/dawn/ContextDawn.cpp [hunk 5]
```diff
@@ -180,6 +180,12 @@ bool ContextDawn::initialize(
     glfwGetFramebufferSize(mWindow, &mClientWidth, &mClientHeight);
 
     mInstance = std::make_unique<dawn_native::Instance>();
+
+	// Enable debug layer in Debug mode
+	#if defined(_DEBUG)
+	mInstance->EnableBackendValidation(true);
+	#endif
```
Comment (JiangYizhou) [inline src/aquarium-optimized/dawn/ContextDawn.cpp:187]: Only enable validation in debug mode.

**Pull request merged**